ports/esp32/mpconfigport.h

```diff
@@ -259,6 +259,14 @@ typedef long mp_off_t;
 // board specifics
 #define MICROPY_PY_SYS_PLATFORM "esp32"
 
+// Enable serial over USB/JTAG serial peripheral
+#ifndef MICROPY_HW_ESP_USB_SERIAL_JTAG
+#define MICROPY_HW_ESP_USB_SERIAL_JTAG (CONFIG_ESP_CONSOLE_USB_SERIAL_JTAG_ENABLED || ( \
+    ESP_IDF_VERSION < ESP_IDF_VERSION_VAL(5, 1, 0) && (         \
+    CONFIG_ESP_CONSOLE_SECONDARY_USB_SERIAL_JTAG ||         \
```

Comment:
I don't think we ever checked/used `CONFIG_ESP_CONSOLE_SECONDARY_USB_SERIAL_JTAG`, so could probably leave that out of this macro definition.  But also doesn't hurt to include it.

Reply:
Actually... maybe the inclusion of this SECONDARY option is why S3 changes behaviour?

Reply:
Yes, this is why S3 changed behaviour. That option is set by default in the ESP-IDF config.

Reply:
So, isn't it possible to just exclude `CONFIG_ESP_CONSOLE_SECONDARY_USB_SERIAL_JTAG` from the check, and then it will work on S3?

Said differently: without this PR, S3 works.  And the only thing that needs fixing is a rename of `CONFIG_ESP_CONSOLE_USB_SERIAL_JTAG_ENABLED` to `CONFIG_ESP_CONSOLE_USB_SERIAL_JTAG` for IDF < 5.1.  And the latter can be achieved probably with:
```c
#if CONFIG_ESP_CONSOLE_USB_SERIAL_JTAG
#define CONFIG_ESP_CONSOLE_USB_SERIAL_JTAG_ENABLED 1
#endif
```

Reply:
I think that suggested patch is actually equivalent behaviour-wise to the first commit in this PR, although I didn't fully understand that until just now when I went to look into it again:

Serial over S3 native USB port currently works on master in two different ways depending on which ESP-IDF version you have, with <5.1 you get native USB and with >=5.1 you get USB Serial/JTAG peripheral (because both `CONFIG_ESP_CONSOLE_SECONDARY_USB_SERIAL_JTAG` and `CONFIG_ESP_CONSOLE_USB_SERIAL_JTAG_ENABLED` are set by default on both S3 and C3). 
That particular regression (switching to the USB/JTAG peripheral) was introduced by me unwittingly.

With the patch (either the three line one or the first commit in this PR by itself) then S3 would become USB Serial/JTAG regardless of ESP-IDF version.

So I think the three line patch would indeed fix the immediate issue with C3 on v5.0.4 and wouldn't make anything else obviously worse. Maybe that is worth doing now as a quick PR to unbreak C3 on nightly builds.

However, it also seems pretty clear that we should clean this up so it's easier to reason about. Either with the approach in this PR, or with something else.